addons/mail/controllers/discuss/channel.py

```diff
@@ -46,10 +45,6 @@ def _process_request_for_all(self, store: Store, name, params):
         if name == "discuss.channel":
             channels = request.env["discuss.channel"].search([("id", "in", params)])
             request.update_context(channels=request.env.context["channels"] | channels)
-            if not_found_channels := request.env["discuss.channel"].browse(
-                OrderedSet(int(cid) for cid in params) - OrderedSet(channels.ids)
-            ):
-                store.delete(not_found_channels)
```

Comment:
Note: this was uncalled for, the main issue is that we wrongly created the thread in JS so we had to delete it. Now no longer creating it when not necessary.

Force delete prevents to save state on channel (such as: cannot read) and clears relations even when unexpected.